components/src/dynamo/vllm/handlers.py

```
@@ -351,30 +353,40 @@ async def sleep(self, body: dict) -> dict:
-            except Exception as unreg_err:
-                logger.warning(
-                    f"[Sleep] Failed to unregister endpoint from discovery: {unreg_err}"
-                )
+                # Step 1: Unregister endpoint instance FIRST to stop new requests from arriving
+                if self.generate_endpoint is not None:
+                    try:
+                        await self.generate_endpoint.unregister_endpoint_instance()
+                        logger.info(
+                            "[Sleep] Unregistered endpoint from discovery - worker removed from routing pool"
+                        )
+                    except Exception as unreg_err:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)